examples/cpp/flow_control/PublisherApp.cpp

```diff
@@ -152,28 +158,26 @@ void PublisherApp::on_publication_matched(
 void PublisherApp::run()
 {
     // Publication code
-    FlowControl st;
+    FlowControl msg;
 
     /* Initialize your structure here */
-    int msgsent_fast = 0;
-    int msgsent_slow = 0;
-    while (!is_stopped() && ((samples_ == 0) || ((msgsent_fast < samples_) && (msgsent_slow < samples_))))
+    uint16_t samples_fast = 0;
+    uint16_t samples_slow = 0;
```

Comment:
I think we can do this with only one counter it is increased here and only if both publish call went OK